### PR TITLE
feat(crates-io): introduce crates-io manager

### LINKED ISSUE #2832: Publish gsdk, gcli, gclient to crates-io via CI
### Problem to Solve

- triggered manually for the patch versions
- triggered with release for the minor versions and the major versions

### Possible Solution

match the release tag and follow the version of node-cli

ref #2831 

### Notes

_No response_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #2832 

- [x] list of all packages that need to be published
- [x] sort the packages in queue
- [x] specify the versions of dependencies in Cargo.toml  ( with package `cargo_toml` )
- [x] rename `gear-core-processor` in demand
- [x] publish packages to crates-io
- [x] skip publishing if the package has already been published
- [x] handle crate-io limits for publishing 
- [x] Dispatchable CI

show case: https://github.com/gear-tech/gear/actions/runs/6873627550/job/18693962567

@gear-tech/dev

### CODE REVIEW

**Review by shamilsan — APPROVED** [review]

File: utils/wasm-instrument/Cargo.toml [hunk 1]
```diff
@@ -17,14 +17,12 @@ enum-iterator.workspace = true
 [dev-dependencies]
 wasmparser.workspace = true
 wat.workspace = true
-gear-backend-wasmi.workspace = true
+gear-backend-wasmi = { workspace = true, features = ["std"] }
```
Comment (shamilsan) [inline utils/wasm-instrument/Cargo.toml:20]: We can remove `gear-backend-wasmi` now and not publish it.

Comment (clearloop) [reply utils/wasm-instrument/Cargo.toml:20]: #3199 

`gcli` currently requires this crate for processing metadata xd

**Comment by clearloop** [comment]
requires refactor after removing `gear-backend-wasmi`

**Comment by clearloop** [comment]
show case: https://github.com/gear-tech/gear/actions/runs/6150434368/job/16688374145?pr=3103

**Review by breathx — CHANGES_REQUESTED** [review]
Please decide what to do now: exclude packages from publish until 3222 solved and then append them or solve 3222 first and then continue on this pr

**Comment by clearloop** [comment]
blocked by #3222

**Comment by breathx** [comment]
> blocked by #3222

Blocker issue now fixed

File: sandbox/host/Cargo.toml [hunk 2]
```diff
@@ -18,8 +18,8 @@ environmental = "1.1.3"
 thiserror.workspace = true
 log = { workspace = true, features = ["std"] }
 wasmer = { version = "2.2", features = ["singlepass"] }
+wasmi = { version = "0.13.2", git = "https://github.com/gear-tech/wasmi", branch = "v0.13.2-sign-ext", features = ["virtual_memory"] }
```
[the remark this answers is not in the document]
Comment (clearloop) [reply sandbox/host/Cargo.toml:21]: it is required if we publishing this crate, but for now we don't need to add it anymore, after introducing metadata executor in gcli

**Comment by clearloop** [comment]
I'm current banned by crates.io for publishing too many packages for testing, need to wait for 24 hrs

**Comment by breathx** [comment]
Waiting for #3460
And why was #3461 closed?

**Comment by clearloop** [comment]
> Waiting for #3460 And why was #3461 closed?

I found that dev-dependencies as crates-io packages is not required for publishing, however, #3460 still makes sense since it cleans the deps of `wasm-instrument`

see the latest publish https://github.com/gear-tech/gear/actions/runs/6831922814/job/18582010768, everything still works

File: gcli/tests/cmd/mod.rs [hunk 3]
```diff
@@ -19,7 +19,9 @@
 mod claim;
 mod info;
 mod key;
-mod program;
+// TODO: #3222
```
Comment (breathx) [inline gcli/tests/cmd/mod.rs:22]: Resolve todo?

File: gcli/tests/common/mod.rs [hunk 4]
```diff
@@ -90,10 +89,10 @@ pub fn login_as_alice() -> Result<()> {
     Ok(())
 }
 
-/// Generate program id from code id and salt
-pub fn program_id(bin: &[u8], salt: &[u8]) -> ProgramId {
-    ProgramId::generate_from_user(CodeId::generate(bin), salt)
-}
+// /// Generate program id from code id and salt
```
Comment (breathx) [inline gcli/tests/common/mod.rs:92]: deadcode

File: sandbox/host/Cargo.toml [hunk 5]
```diff
@@ -18,8 +18,8 @@ environmental = "1.1.3"
 thiserror.workspace = true
 log = { workspace = true, features = ["std"] }
 wasmer = { version = "2.2", features = ["singlepass"] }
-wasmer-types = "2.2"
 wasmi = { git = "https://github.com/gear-tech/wasmi", branch = "v0.13.2-sign-ext", features = ["virtual_memory"] }
+wasmer-types = "2.2"
```
Comment (breathx) [inline sandbox/host/Cargo.toml:22]: workspace

File: utils/crates-io/publish.rs [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    // Packages without local dependencies.
+    "actor-system-error",
+    "gear-common-codegen",
+    "gear-core-errors",
+    "gear-wasm-instrument",
+    "gmeta-codegen",
+    "gsdk-codegen",
+    "gsys",
+    // The packages below have local dependencies,
+    // and should be published in order.
```
Comment (breathx) [inline utils/crates-io/publish.rs:27]: I thought that the main reason of publishing anything for us is to publish basic ecosystem libs: gsys, gcore, gstd, gtest, gclient, gear-wasm-builder, isnt it?

File: Cargo.lock [hunk 7]
```diff
@@ -122,9 +122,9 @@ dependencies = [
 
 [[package]]
 name = "aes-gcm"
-version = "0.10.2"
+version = "0.10.3"
```
Comment (breathx) [inline Cargo.lock:125]: Cargo update was called? I see no reason for it

**Review by breathx — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 7 above)
Comment (clearloop) [reply Cargo.lock]: checked out the Cargo.lock of branch `master`

**Review by breathx — APPROVED** [review]
Ok, waiting next pr with publishing of userspace libs

**Pull request merged**